### ISSUE TITLE
Go panic when running convert mbtiles to pmtiles

### ISSUE THREAD

**Issue by keichan34** [protomaps/go-pmtiles#49]
Here are the logs:

```
2023/04/25 05:38:45 convert.go:362: # of addressed tiles:  5147885
2023/04/25 05:38:45 convert.go:363: # of tile entries (after RLE):  3117056
2023/04/25 05:38:45 convert.go:364: # of tile contents:  2840164
panic: runtime error: index out of range [3117056] with length 3117056
goroutine 1 [running]:
github.com/protomaps/go-pmtiles/pmtiles.build_roots_leaves({0xc021480000, 0x2f9000, 0x35e2aa}, 0x1000)
/Users/bdon/workspace/protomaps/go-pmtiles/pmtiles/directory.go:287 +0x31f
github.com/protomaps/go-pmtiles/pmtiles.optimize_directories({0xc021480000, 0x2f9000, 0x35e2aa}, 0x3f81)
/Users/bdon/workspace/protomaps/go-pmtiles/pmtiles/directory.go:310 +0xad
github.com/protomaps/go-pmtiles/pmtiles.finalize(0xc000186400?, 0xc00011a2a0, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
/Users/bdon/workspace/protomaps/go-pmtiles/pmtiles/convert.go:376 +0x30e
github.com/protomaps/go-pmtiles/pmtiles.ConvertMbtiles(0xc00063fdc8?, {0xc000124570, 0x24}, {0xc000045ec0, 0x3b}, 0x0?, 0x4?)
/Users/bdon/workspace/protomaps/go-pmtiles/pmtiles/convert.go:353 +0xfa8
github.com/protomaps/go-pmtiles/pmtiles.Convert(0x0?, {0xc000124570?, 0x24?}, {0xc000045ec0?, 0x3b?}, 0x7?, 0x58?)
/Users/bdon/workspace/protomaps/go-pmtiles/pmtiles/convert.go:110 +0x97
main.main()
/Users/bdon/workspace/protomaps/go-pmtiles/main.go:150 +0xb19
```

I'm currently working on isolating the problem on my end (it could be a malformed mbtiles archive), but at the very least this error should be actionable. The index out of range .. of length is the same number, so maybe there's an off by one bug somewhere? Has anyone run in to this problem?

**Comment by bdon** [comment]
It's an off-by-one error; let me work on fixing this and getting out a release

**Comment by bdon** [comment]
Released 1.7.1 - can you verify this resolves the issue? https://github.com/protomaps/go-pmtiles/releases/tag/v1.7.1

**Comment by keichan34** [comment]
@bdon Works perfectly. Thanks for the quick fix! :100:
